pkg/controller/applicationlayer/applicationlayer_controller.go

```diff
@@ -277,12 +308,71 @@ func updateApplicationLayerWithDefaults(al *operatorv1.ApplicationLayer) {
 
 // validateApplicationLayer validates ApplicationLayer
 func validateApplicationLayer(al *operatorv1.ApplicationLayer) error {
-	lcSpec := al.Spec.LogCollection
 
-	// If ApplicationLayer spec exists then LogCollection should be set.
-	// TODO: when we will have multiple features in future this should change to at least one feature being set
-	if lcSpec == nil {
-		return fmt.Errorf("missing required LogCollection spec in applicationLayer")
+	// If ApplicationLayer spec exists then one of its features should be set.
+	if al.Spec.LogCollection == nil && al.Spec.WebApplicationFirewall == nil {
+		return fmt.Errorf("ApplicationLayer is not configured")
+	}
+
+	return nil
+}
+
+func getUserDefinedCoreRuleset(ctx context.Context, cli client.Client) (*corev1.ConfigMap, error) {
+	ruleset := new(corev1.ConfigMap)
+
+	if err := cli.Get(
+		ctx,
+		types.NamespacedName{
+			Namespace: common.OperatorNamespace(),
+			Name:      applicationlayer.ModSecurityRulesetConfigMapName,
+		},
+		ruleset,
+	); err != nil && !apierrors.IsNotFound(err) {
+		return nil, err
+	} else if apierrors.IsNotFound(err) {
+		return createDefaultCoreRuleset(ctx, cli)
+	}
+
+	return ruleset, nil
+}
+
+func createDefaultCoreRuleset(ctx context.Context, cli client.Client) (*corev1.ConfigMap, error) {
+	ruleset := &corev1.ConfigMap{
+		TypeMeta: metav1.TypeMeta{Kind: "ConfigMap", APIVersion: "v1"},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      applicationlayer.ModSecurityRulesetConfigMapName,
+			Namespace: common.OperatorNamespace(),
+			Labels:    map[string]string{},
+		},
+		Data: make(map[string]string),
+	}
+
+	for filename, dataBase64 := range applicationlayer.ModsecurityCoreRuleSet {
```

Comment:
I clicked resolved by accident. For now it's alright but it would be nice to have an alternative solution to ConfigMap. Have you any suggestions here?

Reply:
The only alternative solution I can think of is a way of splitting this across multiple configmaps. Let's see if @tmjd or @Brian-McM has thought about this at all. 

@bartolini what's the total size of the configmap once it's all put together? You you forsee that size changing substantially as users customize this? 

I suppose worst-case, we ship with a single configmap and extend to support user-provided extension configmaps in some way, but we haven't thought through how that might work yet.

Reply:
@caseydavenport ah, apologies. I was writing comments here and clicking `Resolve` hoping they will stay here.

The size is almost 900kb so below the limit but close.

From what I know - users are more likely to disable/remove/edit some of these files and add a handful of their own - from what I know that would be a typical use case.

This is a very good point you are making here as I didn't even think about the CM size limit here.

Reply:
I did not consider the size of them and potentially hitting a limit. So no prior thoughts from me.